# **Model Selection**


In the first two tutorials you have learned to download various gene sequences from Genbank and align and concatenate them to create a multi-gene dataset. You also heard about differences between file formats and saw how to create the needed formats. For the next tutorial, we are going to use *ModelFinder* to find the best partitioning scheme for our dataset and also the best models for each partition.

First remember that our final dataset, `Concat.fasta` was formed by 4 genes and you saved it as a fasta format. The mitochondrial genome is formed by 13 protein coding genes and 2 which codes for a ribosomal RNA. So in theory you could repeat what you learned here until now to download all of those genes and concatenate them to create your mitogenomic dataset. I did this for you to save time, so go to [Data](https://github.com/leimurillo/Mol_Phyl_2021/blob/main/Data/) and download `3.15genes.fasta`). Now open *Aliview* and open this file. You should have something similar to this:

<p align="center"><img src="https://github.com/leimurillo/Mol_Phyl_2021/blob/main/Tutorials/3.ModelSelection/Aliview7.png" alt="Aliview7" width="900"></p>

Now I want you to save a \*.phy and a \*.nex alignment file with the name **15genes**. Use the options "***Save as Nexus***" and "***Save as Phylip (full names & padded)***". You should have these two files now:

```
15genes.phy
15genes.nex
```

For substution model selection we are going to use the `15genes.phy` and a command file that we are going to create together. 

**Choosing the right substitution model**

We will use ModelFinder [Kalyaanamoorthy et al. 2017](https://www.nature.com/articles/nmeth.4285), implemented in IQTREE2, to determine the best-fit model. ModelFinder chooses the model that minimizes the BIC score (you can also change the selection criteria to AIC or AICc by adding the option -AIC or -AICc, respectively).

You will need a partition file either in NEXUS or RAxML-style formats, like the examples below: 

The RAxML-style partition file may look like:

```
 DNA, part1 = 1-69 
 DNA, part2 = 70-666
```

Nexus-style partition file may look like:

```
#nexus
begin sets;
    charset part1 = 1-69; 
    charset part2 = 70-666;
end;
```

For simplicity and to save time in this tutorial we will use a partitions file saved in **Data** . So, go to the folder [Data](https://github.com/leimurillo/Mol_Phyl_2021/blob/main/Data/) and download `3.Partitions15genes.txt` 

**Determine the best-fit model**

Once we have the parition file, go into IQTREE2 folder by entering in the terminal


``cd folder/iqtree-2.1.3-MacOSX`` (assuming you dowloaded iqtree-2.1.3-MacOSX, and only if iqtree executable was not copied into system search path).


Then we use the following options:

```
iqtree2 -s 15genes.phy -p 3.Partitions15genes.txt -m TESTMERGEONLY -rcluster 10

```

Here we are selecting the best-fit model for the alignment `-s 15genes.phy` by possibly merging partitions `-m TESTMERGEONLY` to reduce over-parameterization and increase model fit. We use a  partition file `-p 3.Partitions15genes.txt` with relaxed clustering `-rcluster 10` at 10% to save time. These settings for model selection will only consider the invariable site and Gamma rate heterogeneity (thus saving computation time).


But if you want to consider a FreeRate heterogeneity model then run:


```
iqtree2 -s 15genes.phy -p Partitions15genes.txt -m MF+MERGE -rcluster 10

```
ModelFinder will implement a greedy strategy with full partition model, subsequentially merging genes and considering a FreeRate heterogeneity model.

Additionally, if you want to restrict model selection to only those model supported by the specific program (Mr Bayes in our case), then run:


```
iqtree2 -s 15genes.phy -p Partitions15genes.txt -m TESTMERGEONLY -mset mrbayes -rcluster 10 

```

If you are in this section, well done! now you have several files with the selected the best-fit model for our dataset.

Questions:

1. How the model substitutions suggested by ModelFinder look like? compare the files generated by different settings.
